src/halmos/sevm.py

```diff
@@ -903,14 +906,36 @@ def halt(
     def is_halted(self) -> bool:
         return self.context.output.data is not None
 
-    def reverted_with(self, expected: ByteVec) -> bool:
-        if not isinstance(self.context.output.error, Revert):
+    def is_panic_of(self, expected_error_codes: set[int]) -> bool:
+        """
+        Check if the error is Panic(k) for any k in the given error code set.
+        An empty set or None will match any error code.
+
+        Panic(k) is encoded as 36 bytes (4 + 32) consisting of:
+            bytes4(keccak256("Panic(uint256)")) + bytes32(k)
+        """
+
+        output = self.context.output
+
+        if not isinstance(output.error, Revert):
+            return False
+
+        error_data = output.data
+        if byte_length(error_data) != 36:
+            return False
+
+        error_selector = error_data[0:4]
+        if error_selector != PANIC_SELECTOR:
```

Comment:
slightly more efficient to do this:

```py
PANIC_SELECTOR = bytes.fromhex("4E487B71")
...
if PANIC_SELECTOR != error_selector.unwrap()
```

this would avoid repeatedly unwrapping PANIC_SELECTOR and quickly return false if error_selector is not `bytes`

Reply:
nice. forgot that bytevec comparison involves unwrap(). fixed in https://github.com/a16z/halmos/pull/382/commits/825770d7e1adf7ef5b443d9cb23c9872189af7ae